### PR TITLE
Add examples for with_collate()

### LINKED ISSUE #96: Add example for `with_collate()`
Maybe showing sorting with the C locale vs sorting in another collation.

Wild caught usage examples in the [GitHub search results for 'with_collate'](https://github.com/search?q=with_collate+user%3Acran&type=Code).

### PR DESCRIPTION
**Pull request by ellessenne**
Fixes #96.

### CODE REVIEW

File: R/collate.R [hunk 1]
```diff
@@ -12,6 +12,15 @@ set_collate <- function(locale) set_locale(c(LC_COLLATE = locale))[[1]]
 #' @template with
 #' @param new `[character(1)]`\cr New collation order
 #' @param .local_envir `[environment]`\cr The environment to use for scoping.
+#' @examples
+#' \dontrun{
+#'
+#' # Modify collation order:
+#' x <- c("bernard", "bérénice", "béatrice", "boris")
+#' withr::with_collate("fr_FR", sort(x))
```
Comment (jimhester) [inline R/collate.R]: You can remove the `withr::` from the example, examples always run with the package already attached.

File: R/collate.R [hunk 2]
```diff
@@ -12,6 +12,15 @@ set_collate <- function(locale) set_locale(c(LC_COLLATE = locale))[[1]]
 #' @template with
 #' @param new `[character(1)]`\cr New collation order
 #' @param .local_envir `[environment]`\cr The environment to use for scoping.
+#' @examples
+#' \dontrun{
```
Comment (jimhester) [inline R/collate.R]: I think we can actually run this, so it probably _doesn't_ need to be in `\dontrun{}`

**Comment by ellessenne** [comment]
Thanks for the feedback @jimhester, I just pushed an update removing `\dontrun` and the `withr::` prefix!

**Comment by jimhester** [comment]
Great work, this package is now better thanks to you!

Thanks!! You are da bomb!

**Pull request merged**